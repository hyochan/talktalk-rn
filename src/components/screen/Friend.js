// @flow
import React, {
  Component,
  useState,
} from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  FlatList,
} from 'react-native';

import UserListItem from '../shared/UserListItem';
import EmptyListItem from '../shared/EmptyListItem';

import { ratio, colors } from '../../utils/Styles';
import { User as Friend } from '../../models/User';
import { ProfileModalContext } from '../../providers/ProfileModalProvider';
import { getString } from '../../../STRINGS';

import styled from 'styled-components/native';

const StyledContainer = styled.View`
  flex: 1;
  background-color: transparent;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

type Props = {
  navigation: any;
};

function Screen(props: Props) {
  const profileModal = React.useContext(ProfileModalContext);
  const [friends, setFriends] = React.useState([
    new Friend(
      'my_uid',
      'hello',
      '',
      'I am fine today',
      false,
    ),
  ]);

  const renderItem = (item: Friend) => {
    console.log('item', item);
    return (
      <UserListItem
        testID='user'
        user={item}
        onPress={() => {
          if (profileModal && profileModal.dispatch) {
            profileModal.dispatch({
              type: 'show-modal',
              payload: { user: item, deleteMode: true },
            });
          }
        }}
      />
    );
  };

  return (
    <StyledContainer>
      <FlatList
        style={{
          alignSelf: 'stretch',
        }}
        contentContainerStyle={
          friends.length === 0
            ? {
              flex: 1,
              alignItems: 'center',
              justifyContent: 'center',
            }
            : null
        }
        keyExtractor={(item, index) => index.toString()}
        data={friends}
        renderItem={({ item }) => renderItem(item)}
        ListEmptyComponent={
          <EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>
        }
      />
    </StyledContainer>
  );
}

export default Screen;
