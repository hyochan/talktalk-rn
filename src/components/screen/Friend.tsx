import React, {
  Component,
  useState,
  useContext,
} from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  FlatList,
} from 'react-native';
import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';

import UserListItem from '../shared/UserListItem';
import EmptyListItem from '../shared/EmptyListItem';

import { User as Friend } from '../../models/User';
import { ProfileModalContext } from '../../providers/ProfileModalProvider';
import { getString } from '../../../STRINGS';

import styled from 'styled-components/native';

const StyledContainer = styled.View`
  flex: 1;
  background-color: transparent;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

interface IProps {
  navigation: any;
}

function Screen(props: IProps) {
  const profileModal = useContext(ProfileModalContext);
  const [friends, setFriends] = React.useState([
    new Friend(
      'my_uid',
      'hello',
      '',
      'I am fine today',
      false,
    ),
  ]);

  const renderItem = (item: Friend) => {
    return (
      <UserListItem
        testID='user'
        user={item}
        onPress={() => {
          if (profileModal) {
            profileModal.dispatch({
              type: 'show-modal',
              payload: { user: item, deleteMode: true },
            });
          }
        }}
      />
    );
  };

  return (
    <StyledContainer>
      <FlatList
        style={{
          alignSelf: 'stretch',
        }}
        contentContainerStyle={
          friends.length === 0
            ? {
              flex: 1,
              alignItems: 'center',
              justifyContent: 'center',
            }
            : null
        }
        keyExtractor={(item, index) => index.toString()}
        data={friends}
        renderItem={({ item }) => renderItem(item)}
        ListEmptyComponent={
          <EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>
        }
      />
    </StyledContainer>
  );
}

export default Screen;
