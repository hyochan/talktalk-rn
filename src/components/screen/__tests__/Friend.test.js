import 'react-native';
import * as React from 'react';
import Friend from '../Friend';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

const props = {
  navigation: {
    navigate: jest.fn(),
  },
};
const component: React.Element<any> = <Friend {...props}/>;

describe('[Friend] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[Friend] interaction', () => {
  let rendered: renderer.ReactTestRenderer;
  let root: renderer.ReactTestInstance;
  let testingLib: any;

  beforeAll(() => {
    rendered = renderer.create(component);
    root = rendered.root;
    testingLib = render(component);
  });

  it('should dispatch [show-modal] when user is pressed', () => {
    const user = testingLib.getByTestId('user');
    fireEvent(user, 'press');
  });
});
