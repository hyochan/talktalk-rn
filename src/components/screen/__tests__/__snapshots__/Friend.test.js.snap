// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Friend] rendering test renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <RCTScrollView
    ListEmptyComponent={
      <Shared
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flex": 1,
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        No content
      </Shared>
    }
    contentContainerStyle={null}
    data={
      Array [
        User {
          "Chatrooms": Array [],
          "chatrooms": undefined,
          "created": undefined,
          "displayName": "hello",
          "friends": undefined,
          "isOnline": false,
          "photoURL": "",
          "statusMsg": "I am fine today",
          "uid": "my_uid",
          "updated": undefined,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "alignSelf": "stretch",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "borderBottomWidth": 1,
                  "borderColor": "rgb(245,245,245)",
                  "flexDirection": "row",
                  "height": 80,
                  "justifyContent": "flex-start",
                  "paddingHorizontal": 20,
                }
              }
            >
              <Unknown
                color="rgb(65,77,107)"
                light={true}
                name="meh"
                size={40}
              />
              <Text
                style={
                  Object {
                    "color": "rgb(65,77,107)",
                    "fontSize": 14,
                    "marginLeft": 12,
                    "width": 100,
                  }
                }
              >
                hello
              </Text>
              <Text
                style={
                  Object {
                    "borderColor": "rgb(233,237,244)",
                    "borderWidth": 0.3,
                    "color": "rgb(65,77,107)",
                    "fontSize": 12,
                    "maxWidth": 134.2,
                    "paddingHorizontal": 8,
                    "paddingVertical": 4,
                    "position": "absolute",
                    "right": 20,
                  }
                }
              >
                I am fine today
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
